jax/numpy/lax_numpy.py

```diff
@@ -897,6 +897,45 @@ def histogram(a, bins=10, range=None, weights=None, density=None):
     counts = counts / bin_widths / counts.sum()
   return counts, bin_edges
 
+@_wraps(np.histogramdd)
+def histogramdd(sample, bins=10, range=None, weights=None, density=None):
+  _check_arraylike("histogramdd", sample)
+  N, D = shape(sample)
+
+  if weights is not None and weights.shape != (N,):
+    raise ValueError("should have one weight for each sample.")
+
+  bin_idx_by_dim = D*[None]
+  nbins = D*[None]
+  bin_edges_by_dim = D*[None]
+  dedges = D*[None]
+
+  for i in builtins.range(D):
+    bin_edges = histogram_bin_edges(sample[:, i], bins[i], range)
+    bin_idx = searchsorted(bin_edges, sample[:, i], side='right')
+    bin_idx = where(sample[:, i] == bin_edges[-1], bin_idx - 1, bin_idx)
+    bin_idx_by_dim[i] = bin_idx
+    nbins[i] = len(bin_edges) + 1
+    bin_edges_by_dim[i] = bin_edges
+    dedges[i] = diff(bin_edges_by_dim[i])
+
+  xy = ravel_multi_index(bin_idx_by_dim, nbins)
+  hist = bincount(xy, weights, minlength=prod(asarray(nbins)))
+  hist = reshape(hist, nbins)
+  hist = lax.convert_element_type(hist, float32)
+  core = D*(slice(1, -1),)
+  hist = hist[core]
+
+  if density:
+    s = sum(hist)
+    for i in builtins.range(D):
+      _shape = ones(D, int32)
+      _shape = ops.index_update(_shape, i, nbins[i] - 2)
```

Comment:
This might be the problematic piece for compilation. try using numpy operations for the shape to avoid pushing this computation to the device.

Reply:
I did some debugging and it looks like the line where the jit error starts happening is 927 i.e. `hist = hist[core]`.  However the stack trace mentions mentions line 923 i.e. `hist = bincount(xy, weights, minlength=prod(asarray(nbins)))`
```
  File "tests/lax_numpy_test.py", line 2323, in <lambda>
    jnp_fun = lambda a, w: jnp.histogramdd(a, bins=bins, weights=_weights(w), density=density)
  File "/home/alexminnaar/jax/jax/numpy/lax_numpy.py", line 923, in histogramdd
    hist = bincount(xy, weights, minlength=prod(asarray(nbins)))
  File "/home/alexminnaar/jax/jax/numpy/lax_numpy.py", line 1520, in bincount
    length = _max(length, minlength)
jax.traceback_util.FilteredStackTrace: jax.core.ConcretizationTypeError: Abstract tracer value encountered where concrete value is expected.

The problem arose with the `bool` function. 
```
Any idea what might be the culprit?

Reply:
I think the issue is with how you're calling `bincount`. See the comment below